modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableColumnHeaderTest.java

```diff
@@ -252,4 +258,76 @@ public void test_resizeColumnToFitContentMaxRow() {
         assertEquals("Width must be equal to initial value",
                 width, column.getWidth(), 0.001);
     }
+
+    /** Row style must affect the required column width */
+    @Test
+    public void test_resizeColumnToFitContentRowStyle() {
+        TableColumn column = tableView.getColumns().get(0);
+
+        tableView.setRowFactory(this::createSmallRow);
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+        double width = column.getWidth();
+
+        tableView.setRowFactory(this::createLargeRow);
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+        assertTrue("Column width must be greater", width < column.getWidth());
+    }
+
+    private TableRow<Person> createSmallRow(TableView<Person> tableView) {
+        TableRow<Person> row = new TableRow<>();
+        row.setStyle("-fx-font: 24 Amble");
+        return row;
+    }
+
+    /** Test resizeColumnToFitContent in the presence of a non-standard row skin */
+    @Test
+    public void test_resizeColumnToFitContentCustomRowSkin() {
+        TableColumn column = tableView.getColumns().get(0);
+
+        tableView.setRowFactory(this::createCustomRow);
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+        double width = column.getWidth();
+        assertTrue(width > 0);
+    }
+
+    private TableRow<Person> createCustomRow(TableView<Person> tableView) {
+        TableRow<Person> row = new TableRow<>() {
+            protected javafx.scene.control.Skin<?> createDefaultSkin() {
```

Comment:
Very minor: `javafx.scene.control.` is not needed

Reply:
You're right. I'll remove this.